stdlib/discreteDistributions.hk

```diff
@@ -0,0 +1,51 @@
+
+def factorial(n nat):
+	product i from 1 to n+1: i
+
+# Limitation: n <= 20
+# If return w/ real2nat, there are rounding errors
+def choose(n nat, k nat):
+	real2nat((product i from 1 to n+1: i)/((product i from 1 to int2nat(n-k+1): i)*(product i from 1 to k+1: i)))
```

Comment:
This is the 'wrong' way to define choose. What you wrote is the mathematical definition, which is a very rotten way of actually computing it.  Please re-implement.

Also, you should never use `real2nat`. This is a sign of a bug.